tests/test_cli.py

```diff
@@ -0,0 +1,167 @@
+from pathlib import Path
+
+import click.testing
+import pytest
+
+from csv2docx import cli
+
+
+default_outpath = Path.cwd() / "output"
+
+
+@pytest.fixture
+def runner():
+    return click.testing.CliRunner()
+
+
+# not using a tmp_path to test default output folder
+def test_basic_succeeds(runner):
+    result = runner.invoke(
+        cli.main,
+        [
+            "--template",
+            "tests/data/example.docx",
+            "--data",
+            "tests/data/example.csv",
+            "--name",
+            "NAME",
+        ],
+    )
+    assert result.exit_code == 0
+
+    # remove standard output files
+    for f in default_outpath.glob("*.docx"):
+        try:
+            f.unlink()
+        except OSError as e:
+            print("Error: %s : %s" % (f, e.strerror))
+
+
+def test_custom_output_succeeds(runner, tmp_path, tmpdir):
+
+    """testing pathlib.Path object as output"""
+    result_path = runner.invoke(
+        cli.main,
+        [
+            "--template",
+            "tests/data/example.docx",
+            "--data",
+            "tests/data/example.csv",
+            "--name",
+            "NAME",
+            "--path",
+            tmp_path,
+        ],
+    )
+    assert result_path.exit_code == 0
+
+    # testing deep non-existent directory
+    result_unknown_path = runner.invoke(
+        cli.main,
+        [
+            "--template",
+            "tests/data/example.docx",
+            "--data",
+            "tests/data/example.csv",
+            "--name",
+            "NAME",
+            "--path",
+            tmp_path / "deep" / "sub" / "folder",
+        ],
+    )
+    assert result_unknown_path.exit_code == 0
+
+    # testing py.path.local object as output
+    # the `tmpdir` provides a temporary directory unique to this test invocation
+    result_dir = runner.invoke(
+        cli.main,
+        [
+            "--template",
+            "tests/data/example.docx",
+            "--data",
+            "tests/data/example.csv",
+            "--name",
+            "NAME",
+            "--path",
+            tmpdir,
+        ],
+    )
+    assert result_dir.exit_code == 0
```

Comment:
About (i), I tried adding `flake8-aaa` to the linter, but I keep getting an error 
> ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
    flake8-aaa from https://files.pythonhosted.org/packages/4d/52/8b023e4aa6d5f2bf668fb2c3bb3a1869a1eecda06ad345a69c8c1e14a8fd/flake8_aaa-0.9.0-py3-none-any.whl#sha256=828436c9746f912e5351c13846fd56071d4114ec91d93455c1424397eb35a8b6
And I don't see `flake8-aaa` being added to the poetry.lock file as a dev dependency (which should be giving it a pinned version). 

Before, these versions were pinned down when running the linter through `nox`, but I guess because I followed the guide to add the wrapper `install_with_constraints()` we need to explicitly install it as a dev dependency to pin down a version in the `poetry.lock` file. After `poetry add flake8-aaa --dev` it works nicely. Let me know if I missed something here.

Reply:
@davidverweij  -- adding it as a `--dev` and updating `def lint(session`) in `noxfile.py` to use the same plugin is the correct approach for installing and using the plugin 👍